heap/heap.cpp

```diff
@@ -0,0 +1,33 @@
+#include "heap.h"
+#include <vector>
+
+void heap::sift_up(int i){
+    if (i==0){
+        return;
+    }
+    if (tree[i/2]<tree[i]){
```

Comment:
i/2 Неверный индекс, можешь проверить на индексе 6 в дереве, заполненном как 1 2 3 4 5 6, у него потомок должен быть 2, а у тебя 3
(i - 1)/2 вернее
```suggestion
    if (tree[i/2] < tree[i]) {
```
Ниже я вижу, что бинарные операторы ты разделяешь пробелами, делай и тут тогда так